packages/localized-utils/src/localized.ts

```diff
@@ -0,0 +1,163 @@
+import invariant from 'tiny-invariant';
+import uniq from 'lodash/uniq';
+import { filterDataAttributes } from '@commercetools-uikit/utils';
+
+type TLocalizedString = {
+  [locale: string]: string;
+};
+type TTouchedLocalizedString = {
+  [locale: string]: boolean;
+};
+type TDataAttributes = 'data-track-component' | 'data-testid' | 'data-test';
+type TLocalizedDataAttributes = Record<TDataAttributes | string, unknown>;
+type TSplittedLanguageGroups = {
+  related: string[];
+  unrelated: string[];
+};
+
+export const getId = (
+  idPrefix?: string,
+  language?: string
+): string | undefined =>
+  idPrefix && language ? `${idPrefix}.${language}` : undefined;
+
+export const getName = (
+  namePrefix?: string,
+  language?: string
+): string | undefined =>
+  namePrefix && language ? `${namePrefix}.${language}` : undefined;
+
+export const getPrimaryLanguage = (language: string): string =>
+  language.split('-')[0];
+
+// splits the languages into two groups:
+//  - the first group starts with the same tag as the selected language
+//  - the second group starts with a different tag
+export const splitLanguages = (
+  selectedLanguage: string,
+  languages: string[]
+): TSplittedLanguageGroups => {
+  const primaryLanguage = getPrimaryLanguage(selectedLanguage);
+  const related = languages.filter(
+    (language) => getPrimaryLanguage(language) === primaryLanguage
+  );
+  const unrelated = languages.filter(
+    (language) => getPrimaryLanguage(language) !== primaryLanguage
+  );
+  return {
+    unrelated,
+    related,
+  };
+};
+
+// sorts the languages with the following priority:
+// - The selected language is placed first (e.g pt-BR)
+// - All languages using the same primary language as the selected language
+//   follow (e.g. pt, pt-PT). They are sorted alphabetically.
+// - All other languages follow, sorted alphabetically as well
+export const sortLanguages = (
+  selectedLanguage: string,
+  allLanguages: string[]
```

Comment:
Just sharing something I struggled with in the past. I always wonder when I should give primitive types names and if that helps. So here `type TLanguage = string` and `TLanguages= TLanguage[]`. It felt silly at first but then I liked it. I don't know. Would love to hear your thoughts.

Reply:
I wanted to do the exact same thing, but decided to skip it.
It felt silly at first, and it still is silly.

Developers should be aware on the context of the variable they work with. The context is given through proper variable naming and accumulated domain knowledge. Because of this, it is redundant to extract type inference into a specified type. (e.g `languages = TLanguages[]` vs `languages = string[]`.

My internal guideline is:

1. Be meticulous about naming. Because of this, specifying primitives should be straight forward (simply `languages = string[]`). Given the reader of the code has some domain knowledge, `languages` must be `string[]`.
2. Specifying type should be considered bonus, and they don't apply on primitives.
3. Do you need to extract the primitive into a specified type?
Unless it is a field on, then you have the problem of mixing concerns.

Reply:
Thanks for taking the discussion. I agree with your points!